### ISSUE TITLE
assert and Extract refactorings

### ISSUE THREAD

**Issue by gavinking** [ceylon/ceylon-ide-eclipse#1422]
Given this code:

```ceylon
void fun(String? string) {
    assert (exists string);
    print(string);
}
```

If I run Extract Function on the `print()` statement, the extracted function has no parameter. It should have a parameter `string`. This is because it mis-identifies the scope of `string`.

**Comment by gavinking** [comment]
Could _possibly_ be related to #1460.

**Comment by davidfestal** [comment]
Indeed this seems to be fixed by the fix of #1460
@gavinking : do you confirm ?

**Comment by gavinking** [comment]
Confirmed, thanks.
